src/m365/pp/commands/tenant/tenant-settings-set.spec.ts

```diff
@@ -0,0 +1,438 @@
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableTrialEnvironmentCreationByNonAdminUsers option is set to true', async () => {
+    const actual = await command.validate({ options: { disableTrialEnvironmentCreationByNonAdminUsers: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableCapacityAllocationByEnvironmentAdmins option is set to false', async () => {
+    const actual = await command.validate({ options: { disableCapacityAllocationByEnvironmentAdmins: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableCapacityAllocationByEnvironmentAdmins option is set to true', async () => {
+    const actual = await command.validate({ options: { disableCapacityAllocationByEnvironmentAdmins: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableSupportTicketsVisibleByAllUsers option is set to false', async () => {
+    const actual = await command.validate({ options: { disableSupportTicketsVisibleByAllUsers: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableSupportTicketsVisibleByAllUsers option is set to true', async () => {
+    const actual = await command.validate({ options: { disableSupportTicketsVisibleByAllUsers: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableDocsSearch option is set to false', async () => {
+    const actual = await command.validate({ options: { disableDocsSearch: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableDocsSearch option is set to true', async () => {
+    const actual = await command.validate({ options: { disableDocsSearch: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableCommunitySearch option is set to false', async () => {
+    const actual = await command.validate({ options: { disableCommunitySearch: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableCommunitySearch option is set to true', async () => {
+    const actual = await command.validate({ options: { disableCommunitySearch: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableBingVideoSearch option is set to false', async () => {
+    const actual = await command.validate({ options: { disableBingVideoSearch: false } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation when the disableBingVideoSearch option is set to true', async () => {
+    const actual = await command.validate({ options: { disableBingVideoSearch: true } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
```

Comment:
This is actually something that we are not testing in our command. If you for example replace a `true` with `"foobar"`, the test will succeed just fine.
This is because we are initializing these options as booleans on minimist, but our validation method doesn't really check this. Therefore I suggest we remove these tests.